src/hooks/useANJ.js

```diff
@@ -21,81 +21,102 @@ import { useConnectedAccount } from '../providers/Web3'
     anjBalanceTypes.Wallet
   )
 
-  const inactiveBalance = useBalanceWithMovements(
-    balances.inactiveBalance,
+  const convertedInactiveBalance = useBalanceWithMovements(
+    inactiveBalance,
     convertedMovements,
     anjBalanceTypes.Inactive
   )
 
-  const activeBalance = useBalanceWithMovements(
-    balances.activeBalance,
+  const convertedActiveBalance = useBalanceWithMovements(
+    activeBalance,
     convertedMovements,
     anjBalanceTypes.Active
   )
 
-  const lockedBalance = useMemo(() => {
-    return { amount: balances.lockedBalance }
-  }, [balances.lockedBalance])
+  const convertedLockedBalance = useMemo(() => {
+    return { amount: lockedBalance }
+  }, [lockedBalance])
 
-  const deactivationBalance = useMemo(() => {
-    return { amount: balances.deactivationBalance }
-  }, [balances.deactivationBalance])
+  const convertedDeactivationBalance = useMemo(() => {
+    return { amount: deactivationBalance }
+  }, [deactivationBalance])
 
   // Since we pass the whole object through props to components, we should memoize it
-  return useMemo(
-    () => ({
-      walletBalance,
-      inactiveBalance,
-      activeBalance,
-      lockedBalance,
-      deactivationBalance,
-    }),
-    [
-      activeBalance,
-      deactivationBalance,
-      inactiveBalance,
-      lockedBalance,
-      walletBalance,
-    ]
-  )
+  return useMemo(() => {
+    if (!balances) {
+      return null
+    }
+
+    return {
+      walletBalance: convertedWalletBalance,
+      inactiveBalance: convertedInactiveBalance,
+      activeBalance: convertedActiveBalance,
+      lockedBalance: convertedLockedBalance,
+      deactivationBalance: convertedDeactivationBalance,
+    }
+  }, [
+    balances,
+    convertedActiveBalance,
+    convertedDeactivationBalance,
+    convertedInactiveBalance,
+    convertedLockedBalance,
+    convertedWalletBalance,
+  ])
 }
 
 // Asummes movements in descending order of creation
 function useConvertedMovements(movements) {
   const { currentTermId } = useCourtClock()
 
-  const effectiveStates = movements.map(mov =>
-    isMovementEffective(mov, currentTermId)
-  )
+  const effectiveStates = movements
+    ? movements.map(mov => isMovementEffective(mov, currentTermId))
+    : []
   const effectiveStatesKey = effectiveStates.join('')
 
   return useMemo(
-    () =>
-      movements.map((mov, i) => ({
+    () => {
+      if (!movements) {
+        return null
+      }
+
+      return movements.map((mov, i) => ({
         ...mov,
         isEffective: effectiveStates[i],
-      })),
+      }))
+    },
     [effectiveStatesKey, movements] //eslint-disable-line
   )
 }
 
 // Calculates the latest movement for each balance
-// In case the balance is the active, we must also calculate all non effective movements to get the effective active balance at current term
+// In case the balance is active or ianctive, we must also calculate all non effective movements to get the effective balance at current term
```

Comment:
```suggestion
// In case the balance is active or inactive, we must also calculate all non effective movements to get the effective balance at current term
```